src/app/helpers/attachment/attachmentLoader.ts

```diff
@@ -0,0 +1,134 @@
+import {
+    binaryStringToArray,
+    concatArrays,
+    decodeBase64,
+    decryptMessage,
+    decryptSessionKey,
+    getMessage
+} from 'pmcrypto';
+
+import { getAttachment } from '../../api/attachments';
+import { MessageExtended } from '../../models/message';
+import { Attachment, AttachmentData } from '../../models/attachment';
+import { AttachmentsCache } from '../../hooks/useAttachments';
+import { Api } from '../../models/utils';
+
+// Reference: Angular/src/app/attachments/services/AttachmentLoader.js
+
+interface SessionKey {
+    data?: Uint8Array;
+    algorithm?: string;
+}
+
+// TODO: Handle isOutside()
+
+export const decrypt = async (
+    encryptedBinaryBuffer: ArrayBuffer,
+    sessionKey: SessionKey = {}
+): Promise<AttachmentData> => {
+    const encryptedBinary = new Uint8Array(encryptedBinaryBuffer);
+
+    try {
+        return await decryptMessage({
+            message: await getMessage(encryptedBinary),
+            sessionKeys: [sessionKey],
+            format: 'binary'
+        });
+    } catch (err) {
+        console.error(err);
+        throw err;
+    }
+};
+
+export const getRequest = ({ ID }: Attachment = {}, api: Api): Promise<ArrayBuffer> => {
+    // if (isOutside()) {
+    //     const decryptedToken = eoStore.getToken();
+    //     const token = $stateParams.tag;
+    //     return Eo.attachment(decryptedToken, token, ID);
+    // }
+
+    return api(getAttachment(ID));
+};
+
+export const getSessionKey = async (attachment: Attachment, message: MessageExtended): Promise<SessionKey> => {
+    // if (attachment.sessionKey) {
+    //     return attachment;
+    // }
+
+    const keyPackets = binaryStringToArray(decodeBase64(attachment.KeyPackets));
+    const options: any = { message: await getMessage(keyPackets) };
+
+    // if (isOutside()) {
+    //     options.passwords = [eoStore.getPassword()];
+    // } else {
+    // options.privateKeys = keysModel.getPrivateKeys(message.AddressID);
+    options.privateKeys = message.privateKeys;
```

Comment:
This is a bit late now but I don't think it's a good idea to cache the private and public keys keys on each message, I think it'd be better to dynamically get them as it's needed and pass them to the functions that need it as arguments

Reply:
I think I will not address that in this PR but I planned a task called something like "keys management".

I must admit it's quite handy to save keys with the message, I could reuse them when I have to deal with attachments later.

But if you think it could be wrong, I'll consider new hooks to get them dynamically.